pkg/api/types.go

```diff
@@ -195,6 +195,119 @@ type VolumeSource struct {
 	NFS *NFSVolumeSource `json:"nfs"`
 }
 
+// Similar to VolumeSource but meant for the administrator who creates PVs.
+// Exactly one of its members must be set.
+type PersistentVolumeSource struct {
+	// GCEPersistentDisk represents a GCE Disk resource that is attached to a
+	// kubelet's host machine and then exposed to the pod.
+	GCEPersistentDisk *GCEPersistentDiskVolumeSource `json:"gcePersistentDisk"`
+	// HostPath represents a directory on the host.
+	// This is useful for development and testing only.
+	// on-host storage is not supported in any way
+	HostPath *HostPathVolumeSource `json:"hostPath"`
+}
+
+type PersistentVolume struct {
+	TypeMeta   `json:",inline"`
+	ObjectMeta `json:"metadata,omitempty"`
+
+	//Spec defines a persistent volume owned by the cluster
```

Comment:
Thanks for so clearly describing your mental model.  It illustrates very well where I think we differ.  Perhaps best discussed in person some time, but I'll dump my brain here in the mean time :-)

In your current design, as I understand it, a PersistentVolumeClaim lays claim to an already existing, mountable volume, which I agree is in the user application domain.  On the other extreme we have what I will call storage pools, for example the EBS service, the GCE PD service, or an on-premise farm of NetApp filers, all of which have API's for creating mountable volumes of specific sizes, performance characteristics etc. Those storage pools are in the infrastructure domain - they are built, capacity planned, maintained etc by cloud providers (or storage administrators in the case of the on-premise NetApp filer farm) and provide the large slicable resource pool to which you refer.  In between these two (volume claims and storage pools) we have the world where the actual volumes are created. I believe this to be logically in the application domain - users should be able to create and delete them on demand.  Right now I think that we have administrators doing that, but I see that as a temporary measure.  It would be worth getting consensus on the answer to this question - can users (in future) create volumes, or only lay claim to volumes that already exist?